doc/manuals/sprokit/embedded_pipeline.rst

```diff
@@ -0,0 +1,327 @@
+*adapter_data_set* available. In the same manner, the `full()` method
+can be called to see if there is space to send a *adapter_data_set*
+before calling `send()`.
+
+
+How to Specify Pipeline
+-----------------------
+
+Pipelines are provided to the *embedded_pipeline* object as a
+stream. The most common types of streams used are file streams and
+string streams. To use a file stream, the controlling program needs to
+open the file and pass the stream to the embedded pipeline
+object. Alternatively, the pipeline can be specified as a string
+stream. The easiest way to build the pipeline definition is to use the
+macros supplied in
+
+``#include <sprokit/pipeline_util/literal_pipeline.h>``
+
+.. doxygendefine:: SPROKIT_PROCESS
+   :project: kwiver
+
+.. doxygendefine:: SPROKIT_CONFIG
+   :project: kwiver
+
+.. doxygendefine:: SPROKIT_CONNECT
+   :project: kwiver
+
+.. doxygendefine:: SPROKIT_CONFIG_BLOCK
+   :project: kwiver
+
+There are additional macros available for more detailed control over
+the pipeline definition. Refer to the full documentation for the details.
+
+If needed, the scheduler type can be specified in the pipeline
+definition as follows: ::
+
+  std::stringstream pipeline_desc;
+  pipeline_desc  << SPROKIT_CONFIG_BLOCK( "_scheduler" )
+                 << SPROKIT_CONFIG( "type", scheduler_type );
+
+
+Advanced Topics
+===============
+
+Overriding Input and/or Output Adapters
+---------------------------------------
+
+There are some cases where the pipeline will directly source its data
+rather than get it from the controlling program. Reading data directly
+from a file is one example. Similarly, there are pipelines that sink
+the output data directly rather than passing it back to the
+controlling program. In both of these cases, the checks for mandatory
+input and output adapter processes need to be bypassed to allow the
+pipeline to run. This is done by deriving a class and overriding the
+`connect_input_adapter()` and/or `connect_output_adapter()` method to
+just return true. The following is an example of overriding the input
+adapter requirement::
+
+  class no_src_embedded_pipeline
+    : public kwiver::embedded_pipeline
+  {
+  public:
+    no_src_embedded_pipeline() { }
+    virtual ~no_src_embedded_pipeline() { }
+
+  protected:
+    virtual bool connect_input_adapter() { return true; }
+   };
+
+
+Modifying the Pipeline Configuration
+------------------------------------
+
+There may be a situation where some part of the pipeline configuration
+must be added or modified at runtime. The `update_config()` method can
+be overridden in a derived class to provide the ability to make
+modifications to the pipeline config prior to building the pipeline.
+
+
+Embedded Pipeline Extentions
+----------------------------
+
+Embedded pipeline extentions (EPX) can be dynamically loaded based on
+the pipeline configuration. One use case for EPX is to make resources
```

Comment:
Missing "sure"... although "make sure" is fairly informal. Given the next sentence, what about: "One use case for EPX is to check resource availability before starting the pipeline, for example, to ensure that there are enough resources for the pipeline to start."